Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerRoleStore.cs

```diff
@@ -118,15 +142,101 @@ public Task<IEnumerable<Role>> GetRoles(string grain, string securableItem = nul
             }
 
             if (!string.IsNullOrWhiteSpace(roleName))
             {
-                roles = roles.Where(p => string.Equals(p.Name, roleName));
+                roles = roles.Where(r => string.Equals(r.Name, roleName));
+            }
+
+            return roles;
+        }
+
+        public async Task<Role> AddPermissionsToRole(Role role, ICollection<Permission> permissions)
+        {
+            var roleEntity = await _authorizationDbContext.Roles
+                .SingleOrDefaultAsync(r =>
+                    r.RoleId == role.Id
+                    && !r.IsDeleted);
+
+            if (role == null)
+            {
+                throw new NotFoundException<Role>($"Could not find {typeof(Role).Name} entity with ID {role.Id}");
+            }
+
+            var permissionIds = permissions.Select(p => p.Id);
+            var permissionDictionary = _authorizationDbContext.Permissions
+                .Where(p => !p.IsDeleted)
+                .Where(p => permissionIds.Contains(p.PermissionId))
+                .ToDictionary(p => p.PermissionId);
+
+            foreach (var permission in permissions)
+            {
+                _authorizationDbContext.RolePermissions.Add(new RolePermission
+                {
+                    RoleId = roleEntity.Id,
+                    PermissionId = permissionDictionary[permission.Id].Id,
+                    PermissionAction = PermissionAction.Allow
+                });
+            }
+
+            await _authorizationDbContext.SaveChangesAsync();
+
+            var updatedRole = await Get(role.Id);
+            return updatedRole;
+        }
+
+        public async Task<Role> RemovePermissionsFromRole(Role role, Guid[] permissionIds)
+        {
+            var roleEntity = await _authorizationDbContext.Roles
+                .Include(r => r.RolePermissions)
+                .SingleOrDefaultAsync(r =>
+                    r.RoleId == role.Id
+                    && !r.IsDeleted);
+
+            if (role == null)
+            {
+                throw new NotFoundException<Role>($"Could not find {typeof(Role).Name} entity with ID {role.Id}");
             }
 
-            return Task.FromResult(roles.Select(p => p.ToModel()).AsEnumerable());
+            var permissionDictionary = _authorizationDbContext.Permissions
+                .Where(p => !p.IsDeleted)
+                .Where(p => permissionIds.Contains(p.PermissionId))
+                .ToDictionary(p => p.PermissionId);
+
+            foreach (var permissionId in permissionIds)
+            {
+                var rolePermissionToRemove = roleEntity.RolePermissions.Single(
+                    rp => rp.RoleId == roleEntity.Id
+                          && rp.PermissionId == permissionDictionary[permissionId].Id);
+
+                roleEntity.RolePermissions.Remove(rolePermissionToRemove);
+            }
+
+            await _authorizationDbContext.SaveChangesAsync();
+
+            var updatedRole = await Get(role.Id);
```

Comment:
Not sure if we need to query to get the role, we already have it.

Reply:
This has been fixed.